### ISSUE TITLE
Protect developers from automatic merge of [event] tags

### ISSUE THREAD

**Issue by ProditorMagnus** [wesnoth/wesnoth#7486]
### Describe the desired feature

Unlike most of WML, [event] contents represents imperative programming statements. When [event] is in [base_unit]-referenced [unit_type] as well as in current [unit_type], events are merged with +tag rules.

This is not what people would expect. #3827 https://github.com/inferno8/wesnoth-Era_of_Magic/issues/44

Some ideas for possible decision
- warning when [event] is found in unit_type which has any kind of parent unit, no matter if because of base_unit, variation, gender or something else
- warning when [event] is found in unit_type which has any kind of parent unit which also has at least one [event]
- warning when [event] is used in unit type merge operation
- silently skip [event] from unit type merge, instead adding all events as they are written

**Comment by Toranks** [comment]
Why not mix without +tag rules? Is it a very complicated solution?

**Comment by gfgtdf** [comment]
The question is: is this something we should fix for 1.16? It could in theory be a source of OOS. Also i think working around it on the wml side is probably not that hard.

**Comment by gfgtdf** [comment]
> Why not mix without +tag rules? Is it a very complicated solution?

The point is here that the oter subtags of [unit] want to be merged with its base units.

**Comment by Pentarctagon** [comment]
I would say this shouldn't go into 1.16.

**Comment by CelticMinstrel** [comment]
> events are merged with +tag rules.

Correction. This is false. Events are not merged with +tag rules (+tag rules always append new tags, unless it's another +tag). They are merged with WML merge rules, as `[set_variables]mode=merge`.

That's why I think a better solution for this issue is to add new syntax to the WML merge rules to prevent certain tags from being merged, so perhaps something like `__append_only=event` might mean that event tags in the current tag should only be appended and never merged.

Whatever is decided, when merging unit types the attribute could be considered implicitly set to avoid merging events, so for the case of unit types it ends up looking exactly the same as Ravana's final proposal, while also adding more control in other places that use merge.

**Comment by ProditorMagnus** [comment]
Updated description to not mention +tag. I am not expecting anything to 1.16, and my impression of stable is that by default nothing should go there.

From my point of view having implicit `__append_only=event` or similar solution would be more useful, but in case there is not enough developer agreement on that, warning would help too.

**Comment by CelticMinstrel** [comment]
True, making it a warning first and then changing the behaviour seems fine to me.
